### PR TITLE
[MAINT] Backend agnostic comparison composition tests

### PR DESCRIPTION
**Pull request by ThomasHepworth**
Some quick tweaks to the comparison composition tests to make them backend agnostic.

### CODE REVIEW

**Comment by ThomasHepworth** [comment]
For testing, you just need to run:
`python3 -m pytest -m all tests/test_comparison_level_composition.py -v`

File: tests/test_comparison_level_composition.py [hunk 1]
```diff
@@ -72,16 +52,47 @@ def test_binary_composition_internals(clause, c_fun):
...
-    )
+    exact_match_sql = f"({q}first_name_l{q} = {q}first_name_r{q}) {clause} ({q}surname_l{q} = {q}surname_r{q})"  # noqa: E501
     assert level.sql_condition == f"NOT ({exact_match_sql})"
 
     with pytest.raises(ValueError):
         c_fun()
 
 
-def test_composition_outputs():
+@mark_with_dialects_excluding("postgres")
```
Comment (ADBond) [inline tests/test_comparison_level_composition.py:62]: Running these locally it seems like they all support postgres without modification, so maybe we can remove these unless there was a particular reason for excluding?

File: tests/test_comparison_level_composition.py [hunk 2]
```diff
@@ -72,16 +52,47 @@ def test_binary_composition_internals(clause, c_fun):
...
+
+
+@mark_with_dialects_excluding("postgres")
+def test_binary_composition_internals_AND(test_helpers, dialect):
+    cll = test_helpers[dialect].cll
+    quo, _ = _get_dialect_quotes(dialect)
+    binary_composition_internals("AND", cll.and_, cll, quo)
+
+
+@mark_with_dialects_including("duckdb")
```
Comment (ADBond) [inline tests/test_comparison_level_composition.py:76]: I wonder if it makes more sense to leave this unmarked - we aren't really testing duckdb itself, but just using it as a tool to check the universal `cll.not_`, and I think we can expect anyone running the tests to have `duckdb` installed as a core requirement. Not a big deal though so happy if you disagree

**Review by ADBond — APPROVED** [review]
Nice one, looks good - just a couple of comments but happy for you to address as you see fit + merge

File: tests/test_comparison_level_composition.py (hunk 1 above)
Comment (ThomasHepworth) [reply tests/test_comparison_level_composition.py]: Thanks Andy - laziness is the answer to your query.

**Pull request merged**